browser/net/brave_ads_status_header_network_delegate_helper.cc

```diff
@@ -22,6 +19,9 @@ int OnBeforeStartTransaction_AdsStatusHeader(
     const ResponseCallback& next_callback,
     std::shared_ptr<BraveRequestInfo> ctx) {
   Profile* profile = Profile::FromBrowserContext(ctx->browser_context);
+  if (!profile || profile->IsOffTheRecord()) {
+    return net::OK;
```

Comment:
nit: would be great to add a high-level comment as to why we return for this use case